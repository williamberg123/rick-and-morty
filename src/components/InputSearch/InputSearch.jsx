import React, { memo } from 'react';

import PropTypes from 'prop-types';

import './InputSearch.css';

function InputSearch({ funcSearch, searchedValue }){
    return (
        <div className="InputSearch">
            <input
                id="inputSearch"
                onChange={funcSearch}
                type="search"
                placeholder="digite aqui para procurar"
                value={searchedValue}
            />
        </div>
    );
}

InputSearch.propTypes = {
    funcSearch: PropTypes.func.isRequired,
    searchedValue: PropTypes.string
};

export default memo(InputSearch);