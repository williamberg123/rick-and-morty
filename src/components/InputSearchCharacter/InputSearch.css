.InputSearch {
    width: 300px;
    max-width: 400px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font-weight: bold;
    margin: 0 10px 0 auto;
    color: rgb(163, 163, 163);
}

.InputSearch input {
    width: 100%;
    height: 40px;
    padding: 10px;
    margin-bottom: 5px;
}

.InputSearch span {
    color: rgb(146, 146, 146);
    margin-left: 3px;
}

.InputSearch-results {
    display: none;
}

@media (max-width: 700px) {
    .InputSearch {
        margin: 10px 0;
    }
}

@media (max-width: 450px) {
    .InputSearch {
        width: 100%;
    }
}