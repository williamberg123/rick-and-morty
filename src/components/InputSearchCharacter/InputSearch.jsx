import React from 'react';

import PropTypes from 'prop-types';

import './InputSearch.css';

export default function InputSearch({ funcSearch, searchedValue }){
    return (
        <div className="InputSearch">
            <input
                id="inputSearch"
                onChange={funcSearch}
                type="search"
                placeholder="digite aqui para procurar"
                value={searchedValue}
            />
        </div>
    );
}

InputSearch.propTypes = {
    funcSearch: PropTypes.func.isRequired,
    searchedValue: PropTypes.string.isRequired
};