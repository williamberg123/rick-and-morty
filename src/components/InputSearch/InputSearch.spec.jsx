import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import InputSearch from './InputSearch';

describe('<InputSearch />', () => {
    it('should render InputSearch correctly', () => {
        const fn = jest.fn();
        render(<InputSearch funcSearch={fn} searchedValue="morty" />);

        expect(screen.getByPlaceholderText(/digite aqui para procurar/i)).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/digite aqui para procurar/i).value).toBe('morty');
    });
    
    it('should call function when is clicked', async () => {
        expect.assertions(2);

        const fn = jest.fn();
        render(<InputSearch funcSearch={fn} />);

        await userEvent.type(screen.getByPlaceholderText(/digite aqui para procurar/i), 'rick');

        expect(screen.getByPlaceholderText(/digite aqui para procurar/i).value).toBe('rick');
        expect(fn).toBeCalledTimes(4);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const { debug } = render(<InputSearch funcSearch={fn} />);

        expect(screen.getByPlaceholderText(/digite aqui para procurar/i).parentElement)
            .toMatchSnapshot();
    });
});

// Testado