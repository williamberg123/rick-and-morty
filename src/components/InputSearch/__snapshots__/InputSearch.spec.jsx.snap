// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch /> should match snapshot 1`] = `
<div
  class="InputSearch"
>
  <input
    id="inputSearch"
    placeholder="digite aqui para procurar"
    type="search"
    value=""
  />
</div>
`;
